### PR TITLE
Pinned jax and tensorflow versions for external-libraries-tests

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:**
Fixes few CI issues that was causing build failures.

- Pinned versions of jax/tensorflow being installed for `external-libraries-tests`
- Removed keras_version input

**Description of the Change:**
As stated in the context, this PR introduces to changes.

`Pinned versions of jax/tensorflow being installed for `external-libraries-tests`
Previously, this was not pinned, causing CI to always installed the latest available version of these packages from Pypi. Which was causing sporradic test failures for the job.

`Removed keras_version input`
THis was an input into `unit-test.yml` .. but it was never used as we always install the same version of tensorflow/keras. Therefore, it'd cause not confusion in the future in having that input.

**Benefits:**
- More deterministic CI builds.

**Possible Drawbacks:**
None.

**Related GitHub Issues:**
None. [Shortcut Story](https://app.shortcut.com/xanaduai/story/48505/fix-incorrect-versions-of-jax-tensorflow-being-installed-for-external-libraries-tests-on-pennylane-ci)

### CODE REVIEW

File: .github/workflows/unit-test.yml
Comment (albi3ro) [inline .github/workflows/unit-test.yml:1]: For the things below that say `Leave empty to install latest version.`. Is the latest version installed or the pinned version?

Comment (rashidnhm) [reply .github/workflows/unit-test.yml:1]: Latest available version on pypi

Ref:
- [TF](https://github.com/PennyLaneAI/pennylane/blob/57c5d074109cf1df11b24cd5f4c964f130f3c0a1/.github/workflows/unit-test.yml#L122)
- [Jax](https://github.com/PennyLaneAI/pennylane/blob/57c5d074109cf1df11b24cd5f4c964f130f3c0a1/.github/workflows/unit-test.yml#L128)
- [Torch](https://github.com/PennyLaneAI/pennylane/blob/57c5d074109cf1df11b24cd5f4c964f130f3c0a1/.github/workflows/unit-test.yml#L116)

Comment (rashidnhm) [reply .github/workflows/unit-test.yml:1]: These packages were being installed separately outside of requirements file prior to all the CI changes. That being said, if we want to use the version in requirements file if a version is not specified .. then that would require a separate story and some development work to implement (but can be done).

**Review by albi3ro — APPROVED** [review]
thanks for this fix 👍

**Review by timmysilv — APPROVED** [review]

**Pull request merged**